### PR TITLE
Recurrent Maintenance Window - API Docs

### PR DESCRIPTION
**Pull request by chunwahchung**
# Why
To update Instana API docs with the Recurrent Maintenance Window work.

# What
- tag description for Maintenance Configuration
- operation descriptions
- payload samples
- response samples

### CODE REVIEW

**Comment by chunwahchung** [comment]
Dan suggested descriptions of RRULE function summary for things like `BYHOUR`

**Comment by chunwahchung** [comment]
line 87 example/use case https://github.ibm.com/instana/docs/pull/3463/files#diff-51529cacfa072b3ae0ed9ce97795bb83873279fbd613e246c41127a96ecc5661R78

**Comment by chunwahchung** [comment]
the pause & resume the `NOTE` double negative

**Comment by chunwahchung** [comment]
update note to say: won't receive alerts based on the scope defined in the maintenance config

File: spec/descriptions/deleteMaintenanceConfig.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+This endpoint deletes a maintenance configuration given its ID.
+
+### Path Parameters:
+
+- **id:** The ID of the maintenance config to fetch.
```
Comment (luciabotoacai) [inline spec/descriptions/deleteMaintenanceConfig.md:5]: **nitpick:** ... config to delete.

File: spec/descriptions/putMaintenanceConfig.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+### Path Parameters:
+
+- **id:** The ID of the maintenance config to create or update.
+
+### Types of Maintenance Configurations
+- One-Time Maintenance Window
+- Recurrent Maintenance Window
+
+### **Recurrent Maintenance Window - RRULE Limitations**
```
Comment (luciabotoacai) [inline spec/descriptions/putMaintenanceConfig.md:11]: I would change the title to `**Recurrent Maintenance Window**`

File: spec/descriptions/putMaintenanceConfig.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    - [YEARLY](https://datatracker.ietf.org/doc/html/rfc5545#:~:text=or%20more%3B%20and-,YEARLY,-%2C%20to%20specify%20repeating)
+        - Example: every year on December 25th
+        - (monthly by day) `BYDAY` → one specified, not multiple
+        - (monthly by date) `BYMONTHDAY` → one specified, not multiple
+- [UNTIL](https://datatracker.ietf.org/doc/html/rfc5545#:~:text=eight%20days.%0A%0A%20%20%20%20%20%20The-,UNTIL,-rule%20part%20defines)
+    - if provided, only UTC specification
+- [INTERVAL](https://datatracker.ietf.org/doc/html/rfc5545#:~:text=September%202009%0A%0A%0A%20%20%20%20%20%20The-,INTERVAL,-rule%20part%20contains)
+    - for `YEARLY`, always 1
+
+#### **Unsupported** ❌
```
Comment (luciabotoacai) [inline spec/descriptions/putMaintenanceConfig.md:34]: I would remove this section, as we can't keep updating it if new stuff is introduced in the RFC spec

File: spec/descriptions/putMaintenanceConfig.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+- **id:** The ID of the maintenance config to create or update.
+
+### Types of Maintenance Configurations
+- One-Time Maintenance Window
+- Recurrent Maintenance Window
+
+### **Recurrent Maintenance Window - RRULE Limitations**
+
+_NOTE: Not all `RRULE` options from the [iCalendar Spec](https://datatracker.ietf.org/doc/html/rfc5545) are supported._
```
Comment (luciabotoacai) [inline spec/descriptions/putMaintenanceConfig.md:13]: I would rephrase this as mention something like this:
Support for scheduling recurrent maintenance configurations is possible using the RRULE standard from the [iCalendar Spec](https://datatracker.ietf.org/doc/html/rfc5545). 
The following RRULE tokens are supported: "FREQ", "UNTIL", "COUNT", "INTERVAL", "BYDAY", "BYMONTHDAY", "BYMONTH". 

Then, just remove the `Supported` title, but keep the content

File: spec/descriptions/putMaintenanceConfig.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+        - (monthly by day) `BYDAY` → one specified, not multiple
+        - (monthly by date) `BYMONTHDAY` → one specified, not multiple
+    - [YEARLY](https://datatracker.ietf.org/doc/html/rfc5545#:~:text=or%20more%3B%20and-,YEARLY,-%2C%20to%20specify%20repeating)
+        - Example: every year on December 25th
+        - (monthly by day) `BYDAY` → one specified, not multiple
+        - (monthly by date) `BYMONTHDAY` → one specified, not multiple
+- [UNTIL](https://datatracker.ietf.org/doc/html/rfc5545#:~:text=eight%20days.%0A%0A%20%20%20%20%20%20The-,UNTIL,-rule%20part%20defines)
+    - if provided, only UTC specification
+- [INTERVAL](https://datatracker.ietf.org/doc/html/rfc5545#:~:text=September%202009%0A%0A%0A%20%20%20%20%20%20The-,INTERVAL,-rule%20part%20contains)
+    - for `YEARLY`, always 1
```
Comment (luciabotoacai) [inline spec/descriptions/putMaintenanceConfig.md:32]: The maximum interval allowed depends on the frequency used: 365 for DAILY, 52 for WEEKLY, 12 for MONTHLY, and 1 for YEARLY

Comment (luciabotoacai) [reply spec/descriptions/putMaintenanceConfig.md:32]: Can we provide the link to the RRULE online tool: https://icalendar.org/rrule-tool.html as a helper to generate RRULES?

File: spec/payload/putMaintenanceConfig.json [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    "id": "maintenanceConfigID",
+    "name": "sampleMaintenanceConfig",
+    "query": "entity.application.id:\"appID\" OR entity.application.id:\"anotherAppID\" ",
+    "scheduling": {
+        "start": 1683827571245,
+        "duration": {
+            "amount": 2,
+            "unit": "HOURS"
+        },
+        "type": "ONE_TIME"
```
Comment (luciabotoacai) [inline spec/payload/putMaintenanceConfig.json:11]: Can we have a payload for recurrence (maybe use `FREQ=WEEKLY;INTERVAL=2;BYDAY=SA;COUNT=10` for the RRULE)

File: spec/descriptions/resumeMaintenanceConfig.md [hunk 7]
```diff
@@ -0,0 +1,7 @@
+This endpoint resumes a maintenance configuration given its ID.
+
+### Path Parameters:
+
+- **id:** The ID of the maintenance config to resume.
+
+_Note: When you resume a maintenance configuration, you will not receive alerts for the scope defined in the maintenance configuration._
```
Comment (luciabotoacai) [inline spec/descriptions/resumeMaintenanceConfig.md:7]: The note here is not correct. Once fixed, can we just move it in the same paragraph as `This endpoint resumes a maintenance configuration given its ID.` ?

File: spec/openapi.yaml
Comment (luciabotoacai) [inline spec/openapi.yaml:1]: We should not commit this file. This will be maintained by Jenkins.

File: spec/descriptions/tagMaintenanceConfiguration.md [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+Maintenance windows allow you to mute alerts.
+
+### Pausing Maintenance Windows
+Maintenance windows can be paused and resumed.
+
+If a maintenance window is paused, alerts will no longer be muted. 
+
+_NOTE: Pausing unmutes alerts. Maintenance Status is separate from pausing a maintenance window._
+
+### Types of Maintenance Configurations
```
Comment (luciabotoacai) [inline spec/descriptions/tagMaintenanceConfiguration.md:11]: Can you move this section before the Pause/Resume one?
Also, can you provide a short description for each? 
One-Time: executed once at the fixed scheduled time, after which they expire
Recurrent: executed on a schedule basis for planned maintenance activities

File: spec/descriptions/tagMaintenanceConfiguration.md [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+- One-Time Maintenance Window
+- Recurrent Maintenance Window
+
+### Maintenance Status
```
Comment (luciabotoacai) [inline spec/descriptions/tagMaintenanceConfiguration.md:15]: I would remove this section, as it is not facing an API customer directly, they would rather care about how to put/pause/resume, rather than checking statuses, and this can also be found in the IBM public docs.

File: spec/descriptions/tagMaintenanceConfiguration.md [hunk 10]
```diff
@@ -0,0 +1,23 @@
+### Maintenance Windows
+Maintenance windows allow you to mute alerts.
+
+### Pausing Maintenance Windows
+Maintenance windows can be paused and resumed.
+
+If a maintenance window is paused, alerts will no longer be muted. 
+
+_NOTE: Pausing unmutes alerts. Maintenance Status is separate from pausing a maintenance window._
```
Comment (luciabotoacai) [inline spec/descriptions/tagMaintenanceConfiguration.md:9]: It seems that in the note we are repeating the previous sentence. I would also not talk about statuses here.

**Review by luciabotoacai — APPROVED** [review]
LGTM 👍

**Comment by neelbhavsar99** [comment]
Thanks for adding the timezoneId change. Overall LGTM 🚀
